r/src/memorypool.cpp

```diff
@@ -19,6 +19,11 @@
 #if defined(ARROW_R_WITH_ARROW)
 #include <arrow/memory_pool.h>
 
+template <typename T>
+struct NoDelete {
+  inline void operator()(T* ptr) {}
+};
+
```

Comment:
```suggestion
```
This can be replaced by a lambda
```c++
std::shared_ptr<arrow::MemoryPool> MemoryPool__default() {
  return std::shared_ptr<arrow::MemoryPool>(arrow::default_memory_pool(),
                                            [](arrow::MemoryPool* not_deleted){});
}
```